q2.java

```diff
@@ -0,0 +1,113 @@
+
+
+//Implementing a linked list
+class LinkedList<T>{
+	Node<T> head;
+	
+	public LinkedList(){
+		head = null;
+	}
+	
+	public void appendLast(T data){
+		Node<T> last = new Node<T>(data);
+		Node<T> current = this.head;
+		if(current == null){
+			head = last;
+			return;
+		}
+		while(current.next!=null){

```

Comment:
What is the complexity of appending elements here. I think it is O(N). So adding elements to the list would get slower the longer the list becomes. Maintaining a pointer to the last element in the list would allow adding to be O(1).